Automattic/wp-calypso
# Redesign My Plans page to use two column layout

Fixes #6758. Uses #7410.

## Testing Instructions

1. Prepare sites with all WPcom and Jetpack plans (can use the same site with cancelling plan subscriptions and purchasing different ones with credits);
2. Navigate to http://calypso.localhost:3000/plans/my-plan/:site;
3. Test whether everything works (features, buttons, wording);
4. Is the layout nice? Try resizing browser window to see how it looks on smaller/bigger devices.

## Screenshots

Note: notice the screen resolutions the screenshots were taken at.

![selection_036](https://cloud.githubusercontent.com/assets/4988512/17786220/1ecf360a-6584-11e6-9e08-170402747d72.jpg)

![selection_037](https://cloud.githubusercontent.com/assets/4988512/17786245/39bebdd2-6584-11e6-9857-1e956b4b0ad8.jpg)

![selection_038](https://cloud.githubusercontent.com/assets/4988512/17786252/3c5ee9e0-6584-11e6-819f-b93dd701aa94.jpg)

![selection_039](https://cloud.githubusercontent.com/assets/4988512/17786256/3eb874a4-6584-11e6-8531-b42aa2495421.jpg)

### Placeholders when loading data

![selection_044](https://cloud.githubusercontent.com/assets/4988512/17788030/00b034dc-658c-11e6-8704-1725ec9985fb.jpg)

I can add more features as placeholders underneath those but I thought this is enough.

TODO/Ideas:

- [x] Group plan header and happiness support features so they are on one row
- [x] Disconnect `PlanPurchaseFeatures` from Redux store and accept props instead (should be more flexible and reusable for `checkout-thank-you` page)
- [x] Have different plan header based on if you are on checkout thank you or My Plans page
- [x] Refactor `current-plan/main` to fully utilize ES6
- [x] Remove `PlanPurchaseFeatures` from devdocs page
- [x] Bring back `current-plan/main` routing
- [x] Needs design tweaks in smaller resolutions
- [x] Add placeholders when data is loading
- [x] Rename `PlanPurchaseFeatures` to `ProductPurchaseFeatures` as plans are not the only products we have
- [x] Move features from `ProductPurchaseFeatures` to separate component called `ProductPurchaseFeaturesList`
- [x] Allow `ProductPurchaseFeatures` to take children from which it'll compose the page (header and features list)
- [x] Fix current plan header two column layout styling issue


cc @gwwar @rralian

Test live: https://calypso.live/?branch=update/my-plans-page-redesign
